interfaces/mount/entry.go

```diff
@@ -192,8 +195,63 @@ func OptsToFlags(opts []string) (flags int, err error) {
 		case "strictatime":
 			flags |= syscall.MS_STRICTATIME
 		default:
-			return 0, fmt.Errorf("unsupported mount option: %q", opt)
+			if !strings.HasPrefix(opt, "x-snapd-") {
+				return 0, fmt.Errorf("unsupported mount option: %q", opt)
+			}
 		}
 	}
 	return flags, nil
 }
+
+// XSnapdMkdirMode returns the file mode associated with x-snapd-mode mount option.
+// If the mode is not specified explicitly then a default mode of 0755 is assumed.
+func (e *Entry) XSnapdMkdirMode() (os.FileMode, error) {
+	var mode os.FileMode = 0755
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd-mkdir-mode=") {
+			kv := strings.SplitN(opt, "=", 2)
+			n, err := fmt.Sscanf(kv[1], "%o", &mode)
+			if err != nil || n != 1 {
+				return mode, fmt.Errorf("cannot parse octal file mode from %q", kv[1])
+			}
+			break
+		}
+	}
+	return mode, nil
+}
+
+// XSnapdMkdirUid returns the user associated with x-snapd-user mount option.  If
+// the mode is not specified explicitly then a default "root" use is
+// returned.
+func (e *Entry) XSnapdMkdirUid() (uid uint64, err error) {
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd-mkdir-uid=") {
+			kv := strings.SplitN(opt, "=", 2)
+			uid, err = osutil.FindUid(kv[1])
+			if err != nil {
+				// The error message is not very useful so just skip it.
+				return uid, fmt.Errorf("cannot resolve user name %q", kv[1])
+			}
+			return uid, nil
+		}
+	}
+	return uid, nil
```

Comment:
I think what you did below in XSnapdMkdirGid returning (0, nil) in such case is better and more readable.